src/app-layer.c

```diff
@@ -99,6 +101,15 @@ static void AppLayerIncFlowCounter(ThreadVars *tv, uint8_t ipproto, AppProto alp
     }
 }
 
+void AppLayerIncTxCounter(ThreadVars *tv, uint8_t ipproto, AppProto alproto,
+                          uint64_t step)
+{
+    if (likely(tv)) {
+        uint8_t ipproto_map = FlowGetProtoMapping(ipproto);
```

Comment:
We can avoid this if the Flow is passed to this function. Then we can use f->protomap